src/core/lib.lua

```diff
@@ -686,6 +686,38 @@ function getenv (name)
    else return nil end
 end
 
+function fstruct(def)
```

Comment:
Maybe we just need the right documentation?
I mean, for sure these are generic functions, not specific to virtio or PCI.